### PR TITLE
[awscontainerinsightreceiver]Public API check

### PR DESCRIPTION
**Pull request by JohnWu20**
**Description:** 
Check the awsecscontainermetricsreceiver to make sure all API is private unless it needs to be public

**Issue:** 
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/4080

### CODE REVIEW

File: receiver/awsecscontainermetricsreceiver/awsecscontainermetrics/utils.go [hunk 1]
```diff
@@ -24,9 +24,9 @@ import (
 	"google.golang.org/protobuf/types/known/timestamppb"
 )
 
-// GenerateDummyMetrics generates two dummy metrics
+// generateDummyMetrics generates two dummy metrics
 // TODO: Remove once this is replaced with actuall metrics generation code
-func GenerateDummyMetrics() *agentmetricspb.ExportMetricsServiceRequest {
+func generateDummyMetrics() *agentmetricspb.ExportMetricsServiceRequest {
```
Comment (bogdandrutu) [inline receiver/awsecscontainermetricsreceiver/awsecscontainermetrics/utils.go:29]: Why is this not in a "_test" file?

Comment (JohnWu20) [reply receiver/awsecscontainermetricsreceiver/awsecscontainermetrics/utils.go:29]: It is tested in the `utils_test.go` file, and I changed it there too

Comment (mxiamxia) [reply receiver/awsecscontainermetricsreceiver/awsecscontainermetrics/utils.go:29]: Hi @JohnWu20 , I think the question is why do we have the dummy data generation code in a non-testing file? Can we remove it or move it into unit test file?

**Comment by mxiamxia** [comment]
I randomly checked some of the files in `awsecscontainermetricsreceiver`, it seems we can make some variables and func private. Eg, could you move `awsecscontainermetrics` folder into a `internal` folder?

https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/awsecscontainermetricsreceiver/awsecscontainermetrics/constant.go

File: receiver/awsecscontainermetricsreceiver/awsecscontainermetrics/utils.go (hunk 1 above)
Comment (JohnWu20) [reply receiver/awsecscontainermetricsreceiver/awsecscontainermetrics/utils.go]: Thanks, removed the function because it is useless.

**Comment by JohnWu20** [comment]
> I randomly checked some of the files in `awsecscontainermetricsreceiver`, it seems we can make some variables and func private. Eg, could you move `awsecscontainermetrics` folder into a `internal` folder?
> 
> https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/receiver/awsecscontainermetricsreceiver/awsecscontainermetrics/constant.go

Thanks @mxiamxia! I have moved `awsecscontainermetrics` folder into a `internal` folder, and have changed the function and variables to not be exported. Please have a look, thanks!

**Review by Aneurysm9 — APPROVED** [review]

**Review by mxiamxia — APPROVED** [review]

**Pull request merged**